internal/groupsuffix/groupsuffix.go

```diff
@@ -101,6 +106,50 @@ func mutateOwnerRefs(replaceFunc func(baseAPIGroup, apiGroupSuffix string) (stri
 	}
 }
 
+func mutateTokenCredentialRequestAuthenticatorAPIGroup(apiGroupSuffix string) func(context.Context, kubeclient.RoundTrip) {
+	return func(_ context.Context, rt kubeclient.RoundTrip) {
+		// we only care if this is a TokenCredentialRequest
+		if rt.Resource() != loginv1alpha1.SchemeGroupVersion.WithResource("tokencredentialrequests") {
```

Comment:
I purposely didn't include a branch for `create` because I was 1) lazy and 2) I couldn't convince myself that it was worth it, since we don't support any other verbs (besides `list`, now, I guess: https://github.com/vmware-tanzu/pinniped/pull/393) and the logic is safe below to handle the case where the `obj` is a `metav1.PartialObjectMetadata{}` (it just exits). I think I see value in checking that it is a `create`, so I can add that branch.

Reply:
Yeah I think it is just safer to be paranoid about any future changes we make and thus filter down as much as possible.

Reply:
Took a pass at this in 0e1336f253f05cace80cc3439540d7aea7782f54.